test/src/main/java/org/apache/accumulo/test/functional/ThriftMaxFrameSizeIT.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.test.functional;
+
+import static org.apache.accumulo.harness.AccumuloITBase.SUNNY_DAY;
+import static org.apache.accumulo.test.functional.ConfigurableMacBase.configureForSsl;
+
+import java.time.Duration;
+
+import org.apache.accumulo.core.client.Accumulo;
+import org.apache.accumulo.core.client.AccumuloClient;
+import org.apache.accumulo.core.conf.Property;
+import org.apache.accumulo.harness.AccumuloClusterHarness;
+import org.apache.accumulo.miniclusterImpl.MiniAccumuloConfigImpl;
+import org.apache.accumulo.server.rpc.ThriftServerType;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.thrift.TConfiguration;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+@Tag(SUNNY_DAY)
+public class ThriftMaxFrameSizeIT extends AccumuloClusterHarness {
+
+  private ThriftServerType serverType;
+
+  @Override
+  protected Duration defaultTimeout() {
+    return Duration.ofMinutes(1);
+  }
+
+  @Override
+  public void configureMiniCluster(MiniAccumuloConfigImpl cfg, Configuration hadoopCoreSite) {
+    cfg.setProperty(Property.GENERAL_RPC_SERVER_TYPE, serverType.name());
+    if (serverType == ThriftServerType.SSL) {
+      configureForSsl(cfg,
+          getSslDir(createTestDir(this.getClass().getName() + "_" + this.testName())));
+    }
+  }
+
+  @Nested
+  class TestDefault extends TestMaxFrameSize {
+    {
+      serverType = ThriftServerType.getDefault();
+    }
```

Comment:
Interesting strategy to use nested test classes to affect the behavior of the `@BeforeEach` method that sets up the minicluster. I think we have some other Thrift tests that could benefit from a similar usage of this. I know there's some proxy tests, at least, that I think got moved to accumulo-proxy repo... but there might be others still here.

However, I think it's a little confusing to use non-static initializer blocks to set the serverType. Can you just change those to explicit no-arg constructors instead?

Reply:
Yeah that's definitely weird and i will definitely fix it, it was probably a by product of me messing around with this test for a long time yesterday and not thinking.

This was my first time trying to do parameterized tests with JUnit 5 and I discovered yesterday there was no way to do parameters for a constructor, JUnit 4 style, which is what I was really trying to do.

So I spent a while just trying to play around with some options so i could use parameters and have them available in the configureMiniCluster method. I was trying all kinds of things before finally coming up with the nested class idea. Anyways, at one point I was messing around with some anonymous classes and had a static initializer so probably just copied it form that and wasn't thinking.

Reply:
Also in regards to your comment about it being useful for other tests, I think so too. Curiously when I was researching online about how to solve this no one really had any answers. Everyones answer was pretty much "you can't do that". I came up with the idea for the nested classes because initially I was thinking I was going to need to write a subclass for every single test which I didn't want to do and then the nested idea hit me.